src/bitpacking.rs

```diff
@@ -58,6 +59,31 @@ pub trait BitPacking: FastLanes {
     unsafe fn unchecked_unpack_single(width: usize, input: &[Self], index: usize) -> Self;
 }
 
+pub trait BitPackingCompare: BitPacking {
+    fn unpack_cmp_impl<const W: usize, F: Fn(Self, Self) -> bool>(
+        input: &[Self; 1024 * W / Self::T],
+        output: &mut [Self; 1024 / Self::T],
+        f: F,
+        eq_value: Self,
+    ) where
+        BitPackWidth<W>: SupportedBitPackWidth<Self>;
+
+    fn unpack_cmp<const W: usize, F: Fn(Self, Self) -> bool>(
+        input: &[Self; 1024 * W / Self::T],
+        output: &mut [u64; 16],
+        comparison: F,
+        eq_value: Self,
+    ) where
+        BitPackWidth<W>: SupportedBitPackWidth<Self>,
+        [(); 1024 / Self::T]:,
+    {
+        let new_output = unsafe {
```

Comment:
You can't do this unsafe cast without any checks

Reply:
added a len check